cmd/oceantv/broadcast/youtube.go

```diff
@@ -103,6 +105,24 @@ func GetService(ctx context.Context, scope string, tokenURI string) (*youtube.Se
 	return s, nil
 }
 
+// AuthChannel checks for a current token under the passed tokenURI, and generates one if it does not
+// yet exist.
+func AuthChannel(ctx context.Context, w http.ResponseWriter, r *http.Request, scope, tokenURI string) error {
+	// Don't regenerate a token if one already exists.
+	_, err := getToken(ctx, tokenURI)
+
+	// TODO: Check if token has expired.
```

Comment:
Don't we need to implement the token expiry check?

Reply:
As it turns out, apparently google refresh tokens won't expire unless revoked by the user. Which I don't think we are able to determine with just the token itself.

Instead, we will just need to handle the error properly in the broadcast manager when we encounter a response indicating that the token has been invalidated.